From e682f6de1c36770b8949a0db99b87d754551b18e Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro snes9x: add 2.68 MHz underclock option

Reported to improve performance for SA-1 titles such as
Kirby's Dream Land 3 and Super Mario RPG
---
 libretro/libretro.cpp            | 6 ++++++
 libretro/libretro_core_options.h | 1 +
 2 files changed, 7 insertions(+)

diff --git a/libretro/libretro.cpp b/libretro/libretro.cpp
index 88d1d533..f51d0e99 100644
--- a/libretro/libretro.cpp
+++ b/libretro/libretro.cpp
@@ -450,6 +450,12 @@ static void update_variables(void)
             Settings.OneSlowClockCycle  = 6;
             Settings.TwoClockCycles     = 12;
         }
+        else if (strcmp(var.value, "8cycles") == 0)
+        {
+            Settings.OneClockCycle      = 8;
+            Settings.OneSlowClockCycle  = 8;
+            Settings.TwoClockCycles     = 12;
+        }
     }
 
     Settings.MaxSpriteTilesPerLine = 34;
diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index d524d362..c75969d9 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -223,6 +223,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "hacks",
       {
          { "disabled",   NULL },
+         { "8cycles",    "2.68 MHz" },
          { "light",      "Light" },
          { "compatible", "Compatible" },
          { "max",        "Max" },
